extensions/s3/src/main/java/io/deephaven/extensions/s3/ResolvingCredentials.java

```diff
@@ -26,20 +26,18 @@
 enum ResolvingCredentials implements AwsSdkV2Credentials {
     INSTANCE;
 
-    private static final AwsCredentialsProviderChain PROVIDER_CHAIN = AwsCredentialsProviderChain.builder()
-            .credentialsProviders(
-                    DefaultCredentialsProvider.create(),
-                    AnonymousCredentialsProvider.create())
-            .reuseLastProviderEnabled(false) // Don't cache because this chain is a shared static instance
-            .build();
-
     @Override
     public final AwsCredentialsProvider awsV2CredentialsProvider(@NotNull final S3Instructions instructions) {
         if (instructions.profileName().isPresent()
                 || instructions.configFilePath().isPresent()
                 || instructions.credentialsFilePath().isPresent()) {
             return ProfileCredentials.INSTANCE.awsV2CredentialsProvider(instructions);
         }
-        return PROVIDER_CHAIN;
+        return AwsCredentialsProviderChain.builder()
+                .credentialsProviders(
+                        DefaultCredentialsProvider.create(),
+                        AnonymousCredentialsProvider.create())
+                .reuseLastProviderEnabled(true)
+                .build();
```

Comment:
Does this mean that if I make two calls to s3, one with creds, then one without, the second call will end up reusing the creds from the previous call?

If so, I'm not sure that's what we want.